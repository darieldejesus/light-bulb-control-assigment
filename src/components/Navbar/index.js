import React, { Component } from 'react';
import moment from 'moment';
import {
  Box,
  Txt,
  Flex,
} from 'rendition';

export default class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      date: this.getDate(),
      time: this.getTime(),
    };
  }

  componentDidMount() {
    this.interval = setInterval(() => this.updateDateTime(), 1000 * 60);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  /**
   * Get the current date i.e Friday 20, July, 2018.
   *
   * @returns {string} Current date.
   */
  getDate = () => moment().format('dddd D, MMMM, YYYY');

  /**
   * Get the current time i.e 10:54pm.
   *
   * @returns {string} Current time.
   */
  getTime = () => moment().format('h:mm a');

  updateDateTime = () => {
    this.setState({
      date: this.getDate(),
      time: this.getTime(),
    });
  };

  render() {
    const { date, time } = this.state;
    return (
      <Box>
        <Flex direction={['column', 'row']} bg="text.dark" alignItems="center">
          <Box width={1} pl={3} flex={2}>
            <Txt.p color="gray.light" fontSize={[0, 1]}>
              {date}
            </Txt.p>
          </Box>
          <Box width={1} flex={1}>
            <Txt.p color="gray.light" align="center" fontSize={[0, 1]}>
              {time}
            </Txt.p>
          </Box>
          <Flex width={1} pr={3} justifyContent="flex-end" flex={2}>
            <Txt.p color="gray.light" fontSize={[0, 1]}>
              Name Surname
            </Txt.p>
          </Flex>
        </Flex>
      </Box>
    );
  }
}
