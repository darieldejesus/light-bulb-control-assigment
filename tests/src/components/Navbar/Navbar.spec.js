import React from 'react';

import Navbar from '../../../../src/components/Navbar';

jest.mock('moment', () => () => ({
  format: () => 'This is supposed to be a time',
}));

describe('Navbar', () => {
  test('Should render the nav bar', () => {
    const wrapper = shallow(<Navbar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Should call "setState" with the new date', () => {
    const wrapper = shallow(<Navbar />);
    wrapper.instance().setState = jest.fn();
    wrapper.instance().updateDateTime();
    expect(wrapper.instance().setState).toBeCalled();
  });
});
