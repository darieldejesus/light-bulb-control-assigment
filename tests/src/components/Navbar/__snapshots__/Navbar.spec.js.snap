// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render the nav bar 1`] = `
<Component>
  <Component
    alignItems="center"
    bg="text.dark"
    direction={
      Array [
        "column",
        "row",
      ]
    }
  >
    <Component
      flex={2}
      pl={3}
      width={1}
    >
      <Component
        color="gray.light"
        fontSize={
          Array [
            0,
            1,
          ]
        }
      >
        This is supposed to be a time
      </Component>
    </Component>
    <Component
      flex={1}
      width={1}
    >
      <Component
        align="center"
        color="gray.light"
        fontSize={
          Array [
            0,
            1,
          ]
        }
      >
        This is supposed to be a time
      </Component>
    </Component>
    <Component
      flex={2}
      justifyContent="flex-end"
      pr={3}
      width={1}
    >
      <Component
        color="gray.light"
        fontSize={
          Array [
            0,
            1,
          ]
        }
      >
        Name Surname
      </Component>
    </Component>
  </Component>
</Component>
`;
